ui-v2/src/components/flow-runs/data-table/data-table.tsx

```diff
@@ -0,0 +1,91 @@
+import type { Deployment } from "@/api/deployments";
+import {
+	type FlowRunWithDeploymentAndFlow,
+	type FlowRunWithFlow,
+} from "@/api/flow-runs";
+import { DataTable } from "@/components/ui/data-table";
+import { StateBadge } from "@/components/ui/state-badge";
+import { TagBadgeGroup } from "@/components/ui/tag-badge-group";
+import {
+	createColumnHelper,
+	getCoreRowModel,
+	getPaginationRowModel,
+	useReactTable,
+} from "@tanstack/react-table";
+import { useMemo } from "react";
+
+import { DeploymentCell } from "./deployment-cell";
+import { NameCell } from "./name-cell";
+
+const columnHelper = createColumnHelper<
+	FlowRunWithDeploymentAndFlow | FlowRunWithFlow
+>();
+
+const createColumns = ({
+	showDeployment,
+}: {
+	showDeployment: boolean;
+}) => {
+	let ret = [
+		columnHelper.display({
+			id: "name",
+			header: "Name",
+			cell: ({ row }) => <NameCell flowRun={row.original} />,
+		}),
+		columnHelper.accessor("state", {
+			id: "state",
+			header: "State",
+			cell: (props) => {
+				const state = props.getValue();
+				if (!state) {
+					return null;
+				}
+				return <StateBadge type={state.type} name={state.name} />;
+			},
+		}),
+		columnHelper.accessor("deployment", {
+			id: "deployment",
+			header: "Deployment",
+			cell: (props) => {
+				const deployment = props.getValue() as Deployment;
+				return <DeploymentCell deployment={deployment} />;
+			},
+		}),
+		columnHelper.accessor("tags", {
+			id: "tags",
+			header: "Tags",
+			cell: (props) => (
+				<TagBadgeGroup tags={props.getValue()} maxTagsDisplayed={4} />
+			),
+		}),
+	];
+	if (!showDeployment) {
+		ret = ret.filter((column) => column.id !== "deployment");
+	}
+	return ret;
```

Comment:
nit but it might a little easier to manage these by conditionally adding them rather than conditionally removing them. Something like
```ts
const response = []

response.push(columnHelper.display(...))
response.push(columnHelper.accessor(...))

if(showDeployment) {
  response.push(columnHelper.accessor(...))
}

return response
```